src/main/kotlin/org/rust/cargo/project/settings/controls/CargoProjectSettingsControl.kt

```diff
@@ -3,61 +3,44 @@ package org.rust.cargo.project.settings.controls
 import com.intellij.openapi.externalSystem.service.settings.AbstractExternalProjectSettingsControl
 import com.intellij.openapi.externalSystem.util.PaintAwarePanel
 import com.intellij.openapi.options.ConfigurationException
-import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ui.configuration.JdkComboBox
+import com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel
+import com.intellij.openapi.util.Conditions
+import org.rust.cargo.project.RustSdkType
 import org.rust.cargo.project.settings.CargoProjectSettings
-import org.rust.cargo.project.settings.controls.builders.CargoProjectSettingsControlBuilder
-import org.rust.cargo.project.settings.controls.builders.CargoProjectSettingsControlBuilderImpl
+import javax.swing.JButton
 
-class CargoProjectSettingsControl(private val myBuilder: CargoProjectSettingsControlBuilder)
-    : AbstractExternalProjectSettingsControl<CargoProjectSettings>(
-        null,
-        myBuilder.getInitialSettings(),
-        myBuilder.getExternalSystemSettingsControlCustomizer()) {
+class CargoProjectSettingsControl(settings: CargoProjectSettings)
+    : AbstractExternalProjectSettingsControl<CargoProjectSettings>(settings) {
 
-    constructor(initialSettings: CargoProjectSettings) : this(CargoProjectSettingsControlBuilderImpl(initialSettings)) {
+    private var sdkModel: ProjectSdksModel = ProjectSdksModel().apply { reset(project) }
+    private var sdkComboBox: JdkComboBox = JdkComboBox(sdkModel, Conditions.equalTo(RustSdkType.INSTANCE)).apply {
+        val button = JButton("Add")
+        setSetupButton(button, project, sdkModel, JdkComboBox.NoneJdkComboBoxItem(), null, false)
     }
 
     override fun fillExtraControls(content: PaintAwarePanel, indentLevel: Int) {
-        myBuilder.createAndFillControls(content, indentLevel)
-    }
-
-    @Throws(ConfigurationException::class)
-    override fun validate(settings: CargoProjectSettings): Boolean {
-        return myBuilder.validate(settings)
-    }
-
-    override fun applyExtraSettings(settings: CargoProjectSettings) {
-        myBuilder.apply(settings)
-    }
-
-    override fun updateInitialExtraSettings() {
-        myBuilder.apply(initialSettings)
-    }
-
-    override fun isExtraSettingModified(): Boolean {
-        return myBuilder.isModified()
+        content.add(sdkComboBox)
+        content.add(sdkComboBox.setUpButton)
```

Comment:
I'd put these two in some kind of panel (or how this is named in Swing, probably [these](https://docs.oracle.com/javase/tutorial/uiswing/layout/visual.html) guys. I have a little WPF background :smile:), which will span for the whole line, so these two controls won't be cluttered in the same line with "project format" option.

Also there should be a label with text `Rust SDK: `, so user will know what does this field mean.

The final layout of these label, combobox and button should be similar to these guys: ![firefox_2016-02-20_17-17-38](https://cloud.githubusercontent.com/assets/3450050/13197720/e6d6f4a2-d7f5-11e5-90f4-e35d884b3631.png)

Also, the option `<No SDK>` could be available, just like in Java's JDK field (you can see in plugin's project structure window what I mean): for example thanks to this, user will be able to open project when his SDK is installing yet.

I have used [this](https://upsource.jetbrains.com/idea-ce/file/idea-ce-5c958ce15d24841a7bc1d29cc99771efaf667790/plugins/gradle/src/org/jetbrains/plugins/gradle/service/settings/IdeaGradleProjectSettingsControlBuilder.java?nav=9916:9942:focused) as a reference to get some understanding what's going on here.